repo/fsrepo/migrations/httpfetcher.go

```diff
@@ -18,43 +17,45 @@ const (
 
 // HttpFetcher fetches files over HTTP
 type HttpFetcher struct {
-	gateway  string
-	distPath string
-	limit    int64
+	distPath  string
+	gateway   string
+	limit     int64
+	userAgent string
 }
 
 var _ Fetcher = (*HttpFetcher)(nil)
 
 // NewHttpFetcher creates a new HttpFetcher
-func NewHttpFetcher() *HttpFetcher {
-	return &HttpFetcher{
-		gateway:  defaultGatewayURL,
+//
+// Specifying "" for distPath sets the default IPNS path.
+// Specifying "" for gateway sets the default.
+// Specifying 0 for fetchLimit sets the default, -1 means no limit.
+func NewHttpFetcher(distPath, gateway, userAgent string, fetchLimit int64) *HttpFetcher {
```

Comment:
This is less lines of code than the functional options approach, although it comes at the cost of needing to break the function definition if we add any more options. Also, by returning `*HttpFetcher` instead of `(*HttpFetcher, error)` we pass the error checking into the call to `Fetch`.

I like the options pattern + errors despite the verbosity because you almost never have to break the interface. However, given that this likely to be used mostly by go-ipfs, ipfs-update, and fs-repo-migrations and the struct is simple enough a consumer could probably just create their own maybe simple, but breakable, is ok here.

Reply:
I think the simpler approach is OK in this case, so will leave it as is.